test/projects.e2e-spec.ts

```diff
@@ -0,0 +1,108 @@
+import { HttpService, INestApplication } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import * as base64 from 'base-64';
+import { empty } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import session = require('supertest-session');
+import { AppModule } from '../src/app.module';
+import { describeIf } from '../src/utils/test-utils';
+import { GithubService } from '../src/github/github.service';
+
+function canRunTheseTests(): boolean {
+  return (
+    process.env.TEST_GITHUB_TOKEN != null &&
+    process.env.TEST_GITHUB_USER != null
+  );
+}
+
+describeIf('ProjectsController (e2e)', canRunTheseTests(), () => {
+  let app: INestApplication;
+  let httpService: HttpService;
+  let token: string;
+  let user: string;
+  let testSession: any;
+  let authenticatedSession: any;
+  let githubService: GithubService;
+
+  async function deleteKeyboardsRepo(): Promise<void> {
+    await httpService.delete(`https://api.github.com/repos/${user}/keyboards`, {
```

Comment:
This seems like a dangerous test. Do we need to reinforce that this test must not be run with the developer's real GH account?

Reply:
With the `keyboards` repo renamed `test_kdo_keyboards` this should be save now.